rtbkit/core/router/filters/static_filters.cc

```diff
@@ -82,6 +82,17 @@ applyExchangeFilter(FilterState& state, const ConfigSet& result) const
 
 void
 SegmentsFilter::
+fillFilterReasons(FilterState& state, ConfigSet& beforeFilt,
+                  ConfigSet& afterFilt, const std::string & segment) const {
+
+    // Some Magic to get all the filtered out configs by this segment.
+    FilterState::FilterReasons& reasons = state.getFilterReasons();
+    reasons[segment] = beforeFilt ^ (beforeFilt & afterFilt);
```

Comment:
I don't think this is quite right. Since `afterFIlt` is a subset of `beforeFilt`, the `beforeFilt & afterFilt` statement will always return `afterFilt`. I believe `beforeFilt ^ afterFilt` is sufficient because that will return the configs that were flipped during filtering and since configs can only go from 1 to 0, this means that we end up with all the configs that were filtered out.